include/interfaces/iHallEffect.hpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Author: Mihai Stefanescu <mihai.stefanescu@rinftech.com>
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#pragma once
+
+namespace upm
+{
+/**
+ * @brief Interface for Hall Effect Sensors
+*/
+  class iHallEffect
+  {
+  public:
+    virtual ~iHallEffect() {}
+    virtual bool magnetDetected() = 0;
```

Comment:
Do we want to encompass linear hall effect sensors?

Reply:
By linear hall effect sensor I'm guessing you mean one in which the output voltage linearly depends on magnetic flux density, meaning the output will be a value in a range.

I don't know how you would group those together?
We could have the magnetDetected() method return type be int. And have the range for linear sensors be (0,1).
And for the threshold sensors the output would be just be 0 or 1 (the MIN and MAX of the range).

Maybe also change the name to something like hallEffect() or magnet().

Does this sound ok?
Will this be confusing for a developer if he is using both types of sensors in his code?

Reply:
I'm fine with iHallEffect being a boolean interface for now.